save-demo/src/main/kotlin/com/saveourtool/save/demo/service/KubernetesService.kt

```diff
@@ -0,0 +1,141 @@
+import com.saveourtool.save.demo.utils.*
+import com.saveourtool.save.utils.*
+
+import io.fabric8.kubernetes.api.model.*
+import io.fabric8.kubernetes.client.KubernetesClient
+import io.ktor.client.*
+import io.ktor.client.engine.apache.*
+import io.ktor.client.plugins.contentnegotiation.*
+import io.ktor.client.request.*
+import io.ktor.http.*
+import io.ktor.serialization.kotlinx.json.*
+import org.slf4j.LoggerFactory
+import org.springframework.http.HttpStatus
+import org.springframework.stereotype.Service
+import org.springframework.web.server.ResponseStatusException
+import reactor.core.publisher.Mono
+import reactor.kotlin.core.publisher.toMono
+
+import java.net.ConnectException
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+import kotlinx.serialization.json.Json
+
+/**
+ * Service for interactions with kubernetes
+ */
+@Service
+class KubernetesService(
+    private val kc: KubernetesClient,
+    private val configProperties: ConfigProperties,
+) {
+    private val kubernetesSettings = configProperties.kubernetes
+
+    /**
+     * @param demo demo entity
+     * @param version version of [demo] that should be run in kubernetes pod
+     * @return [Mono] of [StringResponse] filled with readable message
+     */
+    @Suppress("TOO_MANY_LINES_IN_LAMBDA")
+    fun start(demo: Demo, version: String = "manual"): Mono<StringResponse> = Mono.just(demo)
+        .logValue(logger::info) {
+            "Creating job ${jobNameForDemo(it)}..."
+        }
+        .flatMap { requestedDemo ->
+            try {
+                requestedDemo.also { kc.startJob(requestedDemo, kubernetesSettings) }.toMono()
+            } catch (kre: KubernetesRunnerException) {
+                Mono.error(
+                    ResponseStatusException(
+                        HttpStatus.INTERNAL_SERVER_ERROR,
+                        "Could not create job for ${jobNameForDemo(requestedDemo)}",
+                        kre,
+                    )
+                )
+            }
+        }
+        .flatMap { requestedDemo ->
+            deferredToMono {
+                configureDemoAgent(requestedDemo, version)
+            }
+        }
+
+    /**
+     * @param demo demo entity
+     * @return list of [StatusDetails]
+     */
+    fun stop(demo: Demo): List<StatusDetails> {
+        logger.info("Stopping job...")
+        return kc.getJob(demo, configProperties.kubernetes.namespace).delete()
+    }
+
+    /**
+     * @param demo demo entity
+     * @return deferred filled with [DemoStatus]
+     */
+    fun getStatus(demo: Demo) = scope.async {
+        val url = "${getPodUrl(demo)}/alive"
+        val status = try {
+            httpClient.get(url).status
+        } catch (connectException: ConnectException) {
+            null
+        }
+        when {
+            status == null -> DemoStatus.ERROR
+            status.isSuccess() -> DemoStatus.RUNNING
```

Comment:
it's not always started as I got, as I remember you use two codes: 200 --ok and some 2xx -- initializing